pkg/apis/servicecatalog/v1beta1/types.go

```diff
@@ -92,6 +92,59 @@ type ClusterServiceBrokerSpec struct {
 	// can be manually incremented by a user to manually trigger a relist.
 	// +optional
 	RelistRequests int64 `json:"relistRequests"`
+
+	// CatalogRestrictions allows adding restrictions onto Class/Plans on relist.
+	// +optional
+	CatalogRestrictions *ServiceClassCatalogRestrictions `json:"catalogRestrictions,omitempty"`
+}
+
+// *Requirements type strings have a special format similar to Label Selectors,
+// except the catalog supports only a very specific property set per type as
+// documented below.
+// The predicate format is expected to be `<property><conditional><requirement>`
+// Check the *Requirements type definition for which <property> strings will be allowed.
+// <conditional> is allowed to be one of the following: ==, !=, in, notin
+// <requirement> will be a string value if `==` or `!=` are used.
+// <requirement> will be a set of string values if `in` or `notin` are used.
+// Multiple predicates are allowed to be chained with a comma (,)
+
+// ClusterServiceClassRequirements represents a selector for classes used to filter.
+// Allowed property names:
+// name - the value set to ClusterServiceClass.Name
+// externalName - the value set to ClusterServiceClass.Spec.ExternalName
+type ClusterServiceClassRequirements string
+
+// ClusterServicePlanRequirements represents a selector for plans used to filter.
+// Allowed property names:
+// name - the value set to ClusterServiceClass.Name
+// externalName - the value set to ClusterServiceClass.Spec.ExternalName
+type ClusterServicePlanRequirements string
+
+// ServiceClassCatalogRestrictions contains the restrictions used to on catalog re-list.
+// Some examples of this object are as follows:
+//
+// Goal: Only list Services with the externalName of FooService and BarService,
+// Solution: restrictions := ServiceClassCatalogRestrictions{
+// 		ServiceClass: "externalName in (FooService, BarService)",
+// }
+// This is an example of a whitelist on service externalName.
```

Comment:
This is a blacklist not a whitelist, right? It is explicit blocking FooService and BarService.

Reply:
I think this could be interpreted in a couple ways: either "Allow all services that are NOT Foo/BarService", which would be an allowance rule and a whitelist. However, ultimately what you're saying here is "Block FooService and BarService", which sounds like a blacklist to me.

Reply:
The comment may be confusing, I am staying the previous solution is the whitelist. 

Above example is whitelist: "externalName in (FooService, BarService)"

Below example is a blacklist: "externalName notin (FooService, BarService)",

in means the thing must be contained in the following list to pass the predicate.
notin means the thing must not be contained in the following list to pass the predicate.

I hope it is just confusion around the Goal, Solution, and English Description format. I can fix that.

@staebler @eriknelson

Reply:
Or you folks are getting caught up on the restrictions name. In that case, I am open to other names that are allowed to house both white and black lists together, example: "externalName in (Demo),name not in (AABBCC)"

Reply:
@n3wscott this is getting kind of pedantic, but I interpret a blacklist as "allow everything **but** these", compared to a whitelist that is "Allow nothing, **but** pass these". The distinction being a default allowing everything, or nothing. Given that definition, your examples are exactly correct, and the comment is describing a blacklist.

Reply:
@n3wscott OK. The confusion for me was the ordering of the comments. I read it to apply to the following rather than the preceding. And empty line after between each example would have made it less confusing for me.

Reply:
@n3wscott And, for what it's worth, I had no problem understanding the examples. I don't think that there is a need to change the restrictions name.

Reply:
Ok updated to hopefully clear this up.

>this is getting kind of pedantic

:D

@eriknelson we are staying the same thing and we agree. I updated the comment order and added whitespace to clear this up.

Reply:
ack, sounds good.